lib/proxy_fetcher/manager.rb

```diff
@@ -79,19 +91,33 @@ def get!
 
     # Clean current proxy list from dead proxies (that doesn't respond by timeout)
     def cleanup!
-      lock = Mutex.new
+      queue = Queue.new
+
+      proxies.dup.each do |proxy|
```

Comment:
Do we need `proxy.dup` here with a new implementation?

Reply:
Actually, we don't this.